omod/src/main/java/org/openmrs/module/coreapps/page/controller/MergeVisitsPageController.java

```diff
@@ -52,27 +64,53 @@ public Object controller(@RequestParam("patientId") Patient patient,
 
     public String post(@RequestParam("patientId") Patient patient,
                        @RequestParam("mergeVisits") List<Integer> mergeVisits,
-					   @RequestParam(value = "returnUrl", required = false) String returnUrl,
+                       @RequestParam(value = "returnUrl", required = false) String returnUrl,
                        @SpringBean AdtService service,
                        UiUtils ui,
-                       HttpServletRequest request, PageModel model){
+                       HttpServletRequest request, PageModel model) {
 
+        List<String> mergedVisitIds = new ArrayList<String>();
         if (patient.isVoided() || patient.isPersonVoided()) {
             return "redirect:" + ui.pageLink("coreapps", "patientdashboard/deletedPatient",
-					SimpleObject.create("patientId", patient.getId().toString() ));
+                    SimpleObject.create("patientId", patient.getId().toString()));
         }
 
-        if (mergeVisits!=null && mergeVisits.size() > 0 ){
+        if (mergeVisits != null && mergeVisits.size() > 0) {
             Visit mergedVisit = service.mergeConsecutiveVisits(mergeVisits, patient);
-            if (mergedVisit != null){
+            if (mergedVisit != null) {
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE, ui.message("coreapps.task.mergeVisits.success"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
-            }else{
+            } else {
                 request.getSession().setAttribute("emr.errorMessage", ui.message("coreapps.task.mergeVisits.error"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
             }
+
+            for (int i = 1; i < mergeVisits.size(); i++) {
```

Comment:
Yes, because I want to get all visits that were **removed** because they were merged. But the index 0 of the mergedVisits is not removed. Example: Merge visits [visit5,visit8,visit9] at the end, we will only have the visit5, the others will be merged inside visit5.

Reply:
I will write a comment saying that we want all visits but not the first one.